src/lib.rs

```diff
@@ -376,6 +551,99 @@ impl Contract {
             .create_community_account(new_community.handle.clone())
     }
 
+    #[payable]
+    pub fn edit_proposal(
+        &mut self,
+        id: ProposalId,
+        body: VersionedProposalBody,
+        labels: HashSet<String>,
+    ) {
+        near_sdk::log!("edit_proposal");
+        require!(
+            self.is_allowed_to_edit_proposal(id, Option::None),
+            "The account is not allowed to edit this proposal"
+        );
+        let editor_id = env::predecessor_account_id();
+        let mut proposal: Proposal =
+            self.proposals.get(id.try_into().unwrap()).unwrap_or_else(|| panic!("Proposal id {} not found", id)).into();
```

Comment:
I don't think we need `try_into` here since `id` is already of type `ProposalId`

```suggestion
            self.proposals.get(id).unwrap_or_else(|| panic!("Proposal id {} not found", id)).into();
```

Reply:
Here id is and index of a vector. And it should be u64. But ProposalId is u32. So I would change it for `id.into()`